sklearn/neighbors/_base.py

```diff
@@ -664,10 +670,7 @@ def _kneighbors_reduce_func(self, dist, start, n_neighbors, return_distance):
         # argpartition doesn't guarantee sorted order, so we sort again
         neigh_ind = neigh_ind[sample_range, np.argsort(dist[sample_range, neigh_ind])]
         if return_distance:
-            if self.effective_metric_ == "euclidean":
-                result = np.sqrt(dist[sample_range, neigh_ind]), neigh_ind
-            else:
-                result = dist[sample_range, neigh_ind], neigh_ind
+            result = dist[sample_range, neigh_ind], neigh_ind
```

Comment:
For reviewers: this change comes with the ones made between L774 and L779 of this file.